server/src/main/java/org/elasticsearch/common/bytes/ReleasableBytesReference.java

```diff
@@ -82,6 +82,20 @@ public ReleasableBytesReference retain() {
         return this;
     }
 
+    public ReleasableBytesReference releasableSlice(int from) {
```

Comment:
This seems to be like a "transfer" ownership operation. It seems to me that it is a bit artificial in that it's primary purpose is to avoid an incref/decref solely for the leak tracker.

Reply:
I found the whole thing rather artificial before. We would retain the original, then slice and then build a new releasable from the original. This way at least, we're honest about the fact that this is all backed by the same ref-count and all we really did is move the read offset. In the end part of the motivation for this change is that we aren't actually transferring ownership all the time in the two spots that this method is used in.